astarte_device.c

```diff
@@ -1146,5 +1180,7 @@ static esp_err_t mqtt_event_handler(esp_mqtt_event_handle_t event)
             // Handle MQTT_EVENT_ANY introduced in esp-idf 3.2
             break;
     }
+#if !(ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(5, 0, 0))
```

Comment:
Maybe here is better to avoid negative logic,  evaluate the condition directly with `<`. 
``` c
#if (ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 0))
```